uniffi_bindgen/src/interface/mod.rs

```diff
@@ -1053,33 +1063,119 @@ impl APIConverter<Field> for weedle::dictionary::DictionaryMember<'_> {
         if let Type::Object(_) = type_ {
             bail!("Objects cannot currently appear in record fields");
         }
+        let default = match self.default {
+            None => None,
+            Some(v) => Some(v.value.convert(ci, &type_)?),
+        };
         Ok(Field {
             name: self.identifier.0.to_string(),
             type_,
             required: self.required.is_some(),
-            default: match self.default {
-                None => None,
-                Some(v) => Some(v.value.convert(ci)?),
-            },
+            default,
         })
     }
 }
 
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, Hash)]
+pub enum Radix {
+    Decimal = 10,
+    Octal = 8,
+    Hexadecimal = 16,
+}
+
 // Represents a literal value.
 // Used for e.g. default argument values.
 #[derive(Debug, Clone, Serialize, Deserialize, Hash)]
 pub enum Literal {
     Boolean(bool),
     String(String),
-    // TODO: more types of literal
+    UInt(u64, Radix, Type),
+    Int(i64, Radix, Type),
+    Float(String, Type),
+    Enum(String, Type),
+    EmptySequence,
+    EmptyMap,
+    Null,
 }
 
-impl APIConverter<Literal> for weedle::literal::DefaultValue<'_> {
-    fn convert(&self, _ci: &ComponentInterface) -> Result<Literal> {
-        Ok(match self {
-            weedle::literal::DefaultValue::Boolean(b) => Literal::Boolean(b.0),
-            weedle::literal::DefaultValue::String(s) => Literal::String(s.0.to_string()),
-            _ => bail!("no support for {:?} literal yet", self),
+impl TypedAPIConverter<Literal> for weedle::literal::DefaultValue<'_> {
```

Comment:
Since you only `impl` this for a single type, I wonder if it would be simpler to have `convert_default_value` as a standalone function to begin with, rather than making it a trait. Or, do you anticipate adding more `impl`s for this shortly?

Reply:
I'm not anticipating any additional implementations. I've made the change to a standalone function as suggested.